src/sentry/static/sentry/app/components/externalIssues/abstractExternalIssueForm.tsx

```diff
@@ -220,6 +233,20 @@ export default class AbstractExternalIssueForm<
     };
   };
 
+  getCleanedFields = (): IssueConfigField[] => {
+    const {fetchedFieldOptionsCache, integrationDetails} = this.state;
+
+    const configsFromAPI = (integrationDetails || {})[this.getConfigName()];
+    return (configsFromAPI || []).map(field => {
+      // Overwrite choices from cache.
+      if (fetchedFieldOptionsCache?.hasOwnProperty(field.name)) {
+        field.choices = fetchedFieldOptionsCache[field.name];
```

Comment:
@mgaeta This is actually going to mutate the input state here. We should make a copy of `field` in the `map` before returning it.